extensions/in-memory/assetindex-memory/src/test/java/org/eclipse/dataspaceconnector/metadata/memory/InMemoryAssetIndexLoaderTest.java

```diff
@@ -0,0 +1,88 @@
+package org.eclipse.dataspaceconnector.metadata.memory;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.spi.asset.AssetIndexLoader;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataAddress;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.easymock.EasyMock.niceMock;
+
+// this test aims at testing the thread-safety of the asset index
+public class InMemoryAssetIndexLoaderTest {
+
+    private final Random random = new Random();
+    private AssetIndexLoader assetIndexLoader;
+
+    @Test
+    void insert() {
```

Comment:
This behavior is already covered by the `insert_exists` test, so it could be deleted

Reply:
not quite - the `exists` one verifies that the DataAddress has in fact been replaced.